### PR TITLE
feat(ingest): add Python libs for Urns

### PR DESCRIPTION
**Pull request by tc350981**
In the data model we define Urn as String which make it hard to manipulate. This PR creates python lib for Urn, DatasetUrn, DataPlatformUrn classes which is meant to help interact with Urn easier.
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by shirshanka — CHANGES_REQUESTED** [review]
This is great work and will make working with Urns much nicer. Thanks!
Have a few suggestions for you. 

Also, how do you think we should transition from the `make_x_urn` pattern in `mce_builder.py`, since all connectors use that pathway today.

File: metadata-ingestion/src/datahub/utilities/urns/dataset_urn.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        return DataPlatformUrn.create_from_string(self.get_entity_id()[0])
+
+    def get_table_name(self) -> str:
+        """
+        :return: the dataset name from this DatasetUrn
+        """
+        return self.get_entity_id()[1]
+
+    def get_env(self) -> str:
```
[the remark this answers is not in the document]
Comment (tc350981) [reply metadata-ingestion/src/datahub/utilities/urns/dataset_urn.py:50]: That's a good point. But I am not quite familiar with Python. I tried my best to search online to find a clean way to get all values of FabricTypeClass. But did not make it. In the change I use something like 
```
VALID_FABRIC_SET: Set[str] = set(
        [
            str(getattr(FabricTypeClass, attr)).upper()
            for attr in dir(FabricTypeClass)
            if not callable(getattr(FabricTypeClass, attr)) and not attr.startswith("_")
        ]
    )
```

Please let me know if you have some suggestion about this

**Comment by tc350981** [comment]
> This is great work and will make working with Urns much nicer. Thanks! Have a few suggestions for you.
> 
> Also, how do you think we should transition from the `make_x_urn` pattern in `mce_builder.py`, since all connectors use that pathway today.

maybe we could do it in phases, like:
1. change the implementation of make_x_urn to use x Urn class
2. Use Urn classes in new code
3. Soft deprecate the make_x_urn util function

File: metadata-ingestion/src/datahub/utilities/urns/dataset_urn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shirshanka) [reply metadata-ingestion/src/datahub/utilities/urns/dataset_urn.py:50]: Take a look at this : https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/configuration/source_common.py#L38

[the remark this answers is not in the document]
Comment (tc350981) [reply metadata-ingestion/src/datahub/utilities/urns/dataset_urn.py:50]: @shirshanka ah thanks for replying. Looks like we are using the same logic

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**